OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/BlockedProviderParentRepositoryTests.cs

```diff
@@ -0,0 +1,233 @@
+public class BlockedProviderParentRepositoryTests
+{
+    private DbContextOptions<OutOfSchoolDbContext> dbContextOptions;
+    private List<Parent> parents;
+    private List<Provider> providers;
+    private List<Workshop> workshops;
+    private List<Application> aplications;
+    private List<ChatRoomWorkshop> chatRooms;
+
+    [SetUp]
+    public async Task SetUp()
+    {
+        dbContextOptions = new DbContextOptionsBuilder<OutOfSchoolDbContext>()
+            .UseInMemoryDatabase(databaseName: "OutOfSchoolTestDB")
+            .Options;
+
+        await Seed();
+    }
+
+    [Test]
+    public async Task Block_ShouldAddBlockedProviderParentModelToDatabase()
+    {
+        // Arrange
+        using var context = GetContext();
+        var repository = GetRepository(context);
+        var blockedProviderParent = new BlockedProviderParent
+        {
+            Id = Guid.NewGuid(),
+            ParentId = Guid.NewGuid(),
+            ProviderId = Guid.NewGuid(),
+            Reason = "Test",
+            UserIdBlock = "TestId",
+        };
+
+        var expectedBlockedProviderParentCount = context.BlockedProviderParents.Count() + 1;
+
+        // Act
+        var result = await repository.Block(blockedProviderParent);
+
+        // Assert
+        Assert.AreEqual(expectedBlockedProviderParentCount, context.BlockedProviderParents.Count());
+        Assert.That(result is not null);
+        Assert.AreEqual(result.Id, blockedProviderParent.Id);
+        Assert.That(result.DateTimeTo is null);
+    }
+
+    [Test]
+    public async Task UnBlock_ShouldModifyBlockedProviderParentDateTimeToProperty()
+    {
+        // Arrange
+        using var context = GetContext();
+        var repository = GetRepository(context);
+        var blockedProviderParent = new BlockedProviderParent
+        {
+            Id = Guid.NewGuid(),
+            ParentId = Guid.NewGuid(),
+            ProviderId = Guid.NewGuid(),
+            Reason = "Test",
+            UserIdBlock = "TestId",
+        };
+        var expectedBlockedProviderParentCount = context.BlockedProviderParents.Count() + 1;
+
+        await repository.Block(blockedProviderParent);
+
+        var blockedProviderParentToUnblock = new BlockedProviderParent
+        {
+            Id = blockedProviderParent.Id,
+            ParentId = blockedProviderParent.ParentId,
+            ProviderId = blockedProviderParent.ProviderId,
+            DateTimeTo = DateTime.Now,
+            UserIdUnblock = "TestId",
+        };
+
+        // Act
+        var result = await repository.UnBlock(blockedProviderParentToUnblock);
+
+        // Assert
+        Assert.AreEqual(expectedBlockedProviderParentCount, context.BlockedProviderParents.Count());
+        Assert.That(result is not null);
+        Assert.AreEqual(result.Id, blockedProviderParentToUnblock.Id);
+        Assert.That(result.DateTimeTo is not null);
+    }
+
+    [Test]
+    public async Task Block_ShouldBlockApplicaionAndChatRoomWorkshop()
```

Comment:
Please fix mistype in test names (this and next test) `...Applicaion...`.

Reply:
TY
fixed it

Reply:
Wow, you've found even more places than I had ). Thanks for thorough inspection!